diff --git a/src/backend/access/transam/xlog.c b/src/backend/access/transam/xlog.c
index 1bb1e43042..8a62e4f67d 100644
--- a/src/backend/access/transam/xlog.c
+++ b/src/backend/access/transam/xlog.c
@@ -634,6 +634,14 @@ static XLogwrtResult LogwrtResult = {0, 0};
  * Note: call Reserve/ReleaseExternalFD to track consumption of this FD.
  */
 static int	openLogFile = -1;
+
+#ifdef __PGLITE__
+void
+pgl_xlog_fd_reset(void)
+{
+	openLogFile = -1;
+}
+#endif
 static XLogSegNo openLogSegNo = 0;
 static TimeLineID openLogTLI = 0;
 
diff --git a/src/backend/storage/file/fd.c b/src/backend/storage/file/fd.c
index 601b048a3f..74236bbbd5 100644
--- a/src/backend/storage/file/fd.c
+++ b/src/backend/storage/file/fd.c
@@ -273,6 +273,22 @@ static AllocateDesc *allocatedDescs = NULL;
  */
 static int	numExternalFDs = 0;
 
+#ifdef __PGLITE__
+void
+pgl_fd_reset(void)
+{
+	VfdCache = NULL;
+	SizeVfdCache = 0;
+	nfile = 0;
+	have_xact_temporary_files = false;
+	temporary_files_size = 0;
+	numAllocatedDescs = 0;
+	maxAllocatedDescs = 0;
+	allocatedDescs = NULL;
+	numExternalFDs = 0;
+}
+#endif
+
 /*
  * Number of temporary files opened during the current session;
  * this is used in generation of tempfile names.
