--- a/src/backend/access/transam/xlog.c
+++ b/src/backend/access/transam/xlog.c
@@ -2527,14 +2527,26 @@
 				 * like a checkpoint is needed, forcibly update RedoRecPtr and
 				 * recheck.
 				 */
-#ifndef __PGLITE__				
+#ifdef __PGLITE__
+				{
+					extern int	is_pglite_active;
+
+					if (!is_pglite_active && IsUnderPostmaster &&
+						XLogCheckpointNeeded(openLogSegNo))
+					{
+						(void) GetRedoRecPtr();
+						if (XLogCheckpointNeeded(openLogSegNo))
+							RequestCheckpoint(CHECKPOINT_CAUSE_XLOG);
+					}
+				}
+#else
 				if (IsUnderPostmaster && XLogCheckpointNeeded(openLogSegNo))
 				{
 					(void) GetRedoRecPtr();
 					if (XLogCheckpointNeeded(openLogSegNo))
 						RequestCheckpoint(CHECKPOINT_CAUSE_XLOG);
 				}
-#endif				
+#endif
 			}
 		}
 
--- a/src/backend/postmaster/checkpointer.c
+++ b/src/backend/postmaster/checkpointer.c
@@ -947,7 +947,11 @@
 	/*
 	 * If in a standalone backend, just do it ourselves.
 	 */
-#ifndef __PGLITE__
+#ifdef __PGLITE__
+	extern int	is_pglite_active;
+
+	if (is_pglite_active || !IsPostmasterEnvironment)
+#else
 	if (!IsPostmasterEnvironment)
 #endif
 	{
