--- a/src/include/port/win32_port.h
+++ b/src/include/port/win32_port.h
@@ -489,10 +489,12 @@
 #define bind(s, addr, addrlen) pgwin32_bind(s, addr, addrlen)
 #define listen(s, backlog) pgwin32_listen(s, backlog)
 #define accept(s, addr, addrlen) pgwin32_accept(s, addr, addrlen)
-#define connect(s, name, namelen) pgwin32_connect(s, name, namelen)
 #define select(n, r, w, e, timeout) pgwin32_select(n, r, w, e, timeout)
+#ifndef __PGLITE__
+#define connect(s, name, namelen) pgwin32_connect(s, name, namelen)
 #define recv(s, buf, len, flags) pgwin32_recv(s, buf, len, flags)
 #define send(s, buf, len, flags) pgwin32_send(s, buf, len, flags)
+#endif
 
 extern SOCKET pgwin32_socket(int af, int type, int protocol);
 extern int	pgwin32_bind(SOCKET s, struct sockaddr *addr, int addrlen);
